lib/src/configuration.cc

```diff
@@ -337,6 +375,34 @@ std::string Configuration::setupEventLogLogging(std::string loglevel)
     return loglevel;
 }
 
+std::string Configuration::setupSyslogLogging(std::string loglevel)
+{
+#ifdef _WIN32
+    throw Configuration::Error {
+      lth_loc::format("eventlog is available only on POSIX platforms") };
+#else  // _WIN32
+#ifdef HAS_LTH_SYSLOG
+    std::string facility = HW::GetFlag<std::string>("syslog-facility");
+    lth_log::syslog_facility fac = string_to_syslog_facility(facility);
+
+    // Some syslog implementations won't write to a non-existent logfile,
+    // so create an empty file if it does not exist.
+    if (!fs::exists(logfile_)) {
+        boost::nowide::ofstream file(logfile_.c_str());
+    }
+
```

Comment:
I'm not sure about this one, since you don't really know where the log will be redirected now.
A better place may be where the syslog configures facility->file mapping